### PR TITLE
[CodingStyle] Skip non-empty-string on VarConstantCommentRector

### LINKED ISSUE #7214: `VarConstantCommentRector` changes `non-empty-string` to `string`
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version |  0.13.4           |

`VarConstantCommentRector` replaces `non-empty-string` with `string` unnecessarily and incorrectly.

## Minimal PHP Code Causing Issue

https://getrector.org/demo/504f24c9-957b-4005-80b0-d45882b26407

## Expected Behaviour

Rector should not be changing the type.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7214

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @internalsystemerror

**Comment by internalsystemerror** [comment]
@samsonasik Damn that was fast! Thank you!

File: packages/NodeTypeResolver/TypeAnalyzer/MethodTypeAnalyzer.php [hunk 1]
```diff
@@ -31,9 +30,6 @@ public function isCallTo(MethodCall $methodCall, string $expectedClass, string $
         return $this->isInstanceOf($methodCall->var, $expectedClass);
     }
 
-    /**
-     * @param non-empty-string $expectedName
-     */
```
Comment (TomasVotruba) [inline packages/NodeTypeResolver/TypeAnalyzer/MethodTypeAnalyzer.php]: These should be kept

Comment (samsonasik) [reply packages/NodeTypeResolver/TypeAnalyzer/MethodTypeAnalyzer.php]: Fixed https://github.com/rectorphp/rector-src/pull/2451/commits/02a4a551e7296d9f64b30c59576bb781d732d4a7

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php [hunk 2]
```diff
@@ -92,6 +96,10 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
+        if ($this->shouldSkipStringType($phpDocInfo, $node->consts[0]->value)) {
+            return null;
+        }
```
Comment (TomasVotruba) [inline rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: This should be handled in string type mapper.

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: ok, it previous got ` @param non-empty-string $expectedName` removed on used by other rule, I will check more.

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: updated to check on IdentifierTypeMapper https://github.com/rectorphp/rector-src/pull/2451/commits/aee64c7b05c497ed4fe01923816c0fcc13570446

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: checking in `IdentifierTypeMapper` seems cause removal of `@param` again:

```diff
-    /**
-     * @param non-empty-string $function
-     * @param non-empty-string $class
-     * @param non-empty-string $arrayMethod
-     * @param non-empty-string $nonArrayMethod
-     */
```

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: Let's try register `AccessoryNonEmptyStringType` into `ScalarStringToTypeMapper` https://github.com/rectorphp/rector-src/pull/2451/commits/831a7e3dfd5c5ff308fc20330f513466036c2df6

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: it works 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php (hunk 2 above)
Comment (TomasVotruba) [reply rules/CodingStyle/Rector/ClassConst/VarConstantCommentRector.php]: Thank you, that seems the best place :+1:

**Pull request merged**